dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/program/ProgramStageDataElementServiceTest.java

```diff
@@ -164,4 +186,33 @@ void testRemoveReferencedDataElement() {
     assertThrows(
         DeleteNotAllowedException.class, () -> dataElementService.deleteDataElement(dataElementA));
   }
+
+  @Test
+  void testGetProgramStageDataElementsWithSyncFiltering() {
+    stageDataElementC = new ProgramStageDataElement(singleEvent, dataElementC, false, 3);
+    stageDataElementD = new ProgramStageDataElement(singleEvent, dataElementD, false, 4);
+    stageDataElementC.setSkipSynchronization(true);
+    stageDataElementD.setSkipSynchronization(true);
+
+    programStageDataElementService.addProgramStageDataElement(stageDataElementA);
+    programStageDataElementService.addProgramStageDataElement(stageDataElementB);
+    programStageDataElementService.addProgramStageDataElement(stageDataElementC);
+    programStageDataElementService.addProgramStageDataElement(stageDataElementD);
+
+    Map<String, Set<String>> result =
+        programStageDataElementService.getProgramStageDataElementsWithSkipSynchronizationSetToTrue(
+            programWithOutReg);
+
+    assertNotNull(result);
+    assertTrue(result.containsKey(singleEvent.getUid()));
+
+    Set<String> skipped = result.get(singleEvent.getUid());
+    assertNotNull(skipped);
+    assertEquals(2, skipped.size());
+    assertTrue(skipped.contains(stageDataElementC.getDataElement().getUid()));
+    assertTrue(skipped.contains(stageDataElementD.getDataElement().getUid()));
+
+    assertFalse(skipped.contains(stageDataElementA.getDataElement().getUid()));
+    assertFalse(skipped.contains(stageDataElementB.getDataElement().getUid()));
```

Comment:
Could all of this just be replaced by using `assertContainsOnly`?
```suggestion
    assertContainsOnly(skipped, List.of(stageDataElementC.getDataElement().getUid(), stageDataElementD.getDataElement().getUid()));
    ```